usr/local/www/services_dhcp_relay.php

```diff
@@ -106,106 +125,73 @@
-		</tr>
-		<tr>
-                        <td width="22%" valign="top" class="vncellreq">Enable</td>
-                        <td width="78%" class="vtable">
-			<input name="enable" type="checkbox" value="yes" <?php if ($pconfig['enable']) echo "checked=\"checked\""; ?> onclick="enable_change(false)" />
-                          <strong><?php printf(gettext("Enable DHCP relay on interface"));?></strong>
-			</td>
-		</tr>
-		<tr>
-                        <td width="22%" valign="top" class="vncellreq">Interface(s)</td>
-                        <td width="78%" class="vtable">
-				<select id="interface" name="interface[]" multiple="multiple" class="formselect" size="3">
-			<?php
-                                foreach ($iflist as $ifent => $ifdesc) {
-					if (!is_ipaddr(get_interface_ip($ifent)))
-						continue;
-					echo "<option value=\"{$ifent}\"";
-					if (in_array($ifent, $pconfig['interface']))
-						echo " selected=\"selected\"";
-					echo ">{$ifdesc}</option>\n";
-				}
-			?>
-                                </select>
-				<br />Interfaces without an IP address will not be shown.
-			</td>
-		</tr>
-		<tr>
-	              <td width="22%" valign="top" class="vtable">&nbsp;</td>
-                      <td width="78%" class="vtable">
-<input name="agentoption" type="checkbox" value="yes" <?php if ($pconfig['agentoption']) echo "checked=\"checked\""; ?> />
-                      <strong><?=gettext("Append circuit ID and agent ID to requests"); ?></strong><br />
-                      <?php printf(gettext("If this is checked, the DHCP relay will append the circuit ID (%s interface number) and the agent ID to the DHCP request."), $g['product_name']); ?></td>
-		</tr>
-		<tr>
-                        <td width="22%" valign="top" class="vncellreq"><?=gettext("Destination server");?></td>
-                        <td width="78%" class="vtable">
-                          <input name="server" type="text" class="formfld unknown" id="server" size="20" value="<?=htmlspecialchars($pconfig['server']);?>" />
-                          <br />
-			  <?=gettext("This is the IP address of the server to which DHCP requests are relayed. You can enter multiple server IP addresses, separated by commas.");?>
-                        </td>
-		</tr>
-		<tr>
-                        <td width="22%" valign="top">&nbsp;</td>
-                        <td width="78%">
-                          <input name="Submit" type="submit" class="formbtn" value="<?=gettext("Save");?>" onclick="enable_change(true)" />
-                        </td>
-		</tr>
-	</table>
-	</div>
-    </td>
-  </tr>
-</table>
-</form>
-<script type="text/javascript">
-//<![CDATA[
-enable_change(false);
-//]]>
-</script>
-<?php include("fend.inc"); ?>
-</body>
-</html>
+
+if ($input_errors)
+	print_input_errors($input_errors);
+
+if ($savemsg)
+	print_info_box($savemsg);
+
+$form = new Form;
+
+$section = new Form_Section('DHCP Relay configuration');
+$form->add($section);
```

Comment:
I'm a bit surprised this works, can you move the add to be after all add() calls have been done? Eg. just before ```print $form```